### ISSUE TITLE
Async pipe not working with dotted notation

### ISSUE THREAD

**Issue by saibotma** [dart-lang/angular#1311]
* [2.0.0-dev.53.0] Dart SDK Version
* [5.0.0-alpha] AngularDart Version
* [Windows] Whether you are using Windows, MacOSX, or Linux (if applicable)
* [Chrome, Firefox] Whether you are using Chrome, Safari, Firefox, Edge (if appplicable)

Please look at this Stackoverflow question for more infos, thanks!
https://stackoverflow.com/questions/50269510/angulardart-async-pipe-not-working-with-dotted-notation

**Comment by matanlurey** [comment]
Closing due to lack of reply/action. Please re-open if you still have an issue.

**Comment by leonsenft** [comment]
**TLDR: Working as intended; you've found the solution.**

So my guess is that `Observable.combineLatest2()` returns a new `Stream` instance each invocation, which is exactly the problem I mentioned in my last reply.

Angular is generating code similar to this:

```dart
void detectChanges() {
  var currVal = !(_asyncPipe.transform(viewModel.output.loginButtonEnabled) ?? false);
  if (!identical(_expr, currVal)) {
    _roundedButton.disabled = currVal;
    _expr = currVal;
  }
}
```

Here's what happens:

1. The first change detection cycle, a new `Stream` is returned by `viewModel.output.loginButtonEnabled`. 
2. When the `Stream` resolves with `true`, this will update the `AsyncPipe` so that its next check will return `true`. Anytime an async event happens, this triggers change detection to run again.
3. Now when we reach this code again, if the *same instance* of a stream is passed to `AsyncPipe`, it will return the latest value from that `Stream` which is `true`. **But** your `loginButtonEnabled` getter returns a *new* instance of a `Stream` each time it's called, so it instead will subscribe to that stream, and we're back to step 1.

As you've discovered, the solution is to reuse the same `Stream` instance by caching it in a field on that component.

**Comment by saibotma** [comment]
Thanks for that explanation :)
